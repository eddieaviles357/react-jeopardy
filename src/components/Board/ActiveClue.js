import React from 'react'
import './board.css'

const ActiveClue = ({ content, answer }) => {
  
  const activeClueClickHandler = (evt) => {
    // avoids parents click events to trigger
    evt.stopPropagation()
    const id = evt.target.id
    if(id === 'go-back') { 
      console.log('goback')
    }
    if(id === 'reveal-answer') {
      console.log('reveal-answer')
    }
  }

  const revealAnswer = (evt) => {
    evt.stopPropagation()
    console.log('answer', answer)
  }

  return (
    <div className='active' onClick={activeClueClickHandler}>
      <div className='clue-choice-container'>
        <span id='go-back'>Go Back</span>
        <span id='reveal-answer'>Reveal Answer</span>
      </div>
      { content }
    </div>
  )
}

export default ActiveClue
