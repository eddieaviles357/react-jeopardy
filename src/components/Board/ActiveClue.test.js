'use strict';

import ActiveClue from './ActiveClue'
import { render } from '@testing-library/react'

test('does not crash', () => {
    render(<ActiveClue />);
});

test('Matches snapshot', () => {
    const activeClue = render(<ActiveClue />);
    expect(activeClue.asFragment()).toMatchSnapshot();
});