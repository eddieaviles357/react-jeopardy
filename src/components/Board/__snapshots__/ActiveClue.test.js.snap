// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Active"
    style="transform: translateX(-50%) translateY(-50%) scale(0) translateZ(0);"
  >
    <div
      class="clue-choices-container"
    >
      <span
        id="go-back"
      >
        Go Back
      </span>
      <span
        id="reveal-answer"
      >
        Reveal Answer
      </span>
    </div>
    <hr />
    <div
      class="Active-qa-container"
    >
      <span
        class="Active-content"
      />
      <span
        class="Active-revealed"
      />
    </div>
  </div>
</DocumentFragment>
`;
